crates/workspace/src/pane.rs

```diff
@@ -2067,53 +2073,66 @@ impl Pane {
     }
 
     fn pin_tab_at(&mut self, ix: usize, window: &mut Window, cx: &mut Context<Self>) {
+        self.change_tab_pin_state(ix, PinOperation::Pin, window, cx);
+    }
+
+    fn unpin_tab_at(&mut self, ix: usize, window: &mut Window, cx: &mut Context<Self>) {
+        self.change_tab_pin_state(ix, PinOperation::Unpin, window, cx);
+    }
+
+    fn change_tab_pin_state(
```

Comment:
The `pin_tab_at` and `unpin_tab_at` functions were nearly identical, so I refactored them into one that takes a `PinOperation` variant.